src/sqlfluff/core/templaters/python.py

```diff
@@ -632,44 +632,106 @@ def _split_uniques_coalesce_rest(
+                        if pos > 0:
+                            yield TemplatedFileSlice(
+                                raw_slice.slice_type,
+                                slice(unique_position[0] - pos, unique_position[0]),
+                                slice(unique_position[1] - pos, unique_position[1]),
+                            )
+
+                        # Handle the unique itself and update the bookmark
                         starts = (
-                            unique_position[0] + len(raw_slice.raw),
-                            unique_position[1] + len(raw_slice.raw),
+                            unique_position[0] + len(unq),
+                            unique_position[1] + len(unq),
                         )
                         yield TemplatedFileSlice(
                             raw_slice.slice_type,
-                            # It's a literal so use its length
                             slice(unique_position[0], starts[0]),
                             slice(unique_position[1], starts[1]),
                         )
                         # Move the bookmark after this position
                         bookmark_idx = idx + 1
-                # At the end of the loop deal with any hangover
+
+                        # Handle any remnant after the unique.
+                        if raw_slice.raw[pos + len(unq) :]:
+                            remnant_length = len(raw_slice.raw) - (len(unq) + pos)
+                            _starts = starts
+                            starts = (
+                                starts[0] + remnant_length,
+                                starts[1] + remnant_length,
+                            )
+                            yield TemplatedFileSlice(
+                                raw_slice.slice_type,
+                                slice(_starts[0], starts[0]),
+                                slice(_starts[1], starts[1]),
+                            )
+
+                if bookmark_idx == 0:
+                    # This is a SAFETY VALVE. In Theory we should never be here
+                    # and if we are it implies an error elsewhere. This clause
+                    # should stop any potential infinite recursion in it's tracks
+                    # by simply classifying the whole of the current block as
+                    # templated and just stopping here.
+                    # Bugs triggering this eventuality have been observed in 0.4.0.
+                    templater_logger.info(
+                        "        Safety Value Info: %s, %r",
+                        two_way_uniques,
+                        templated_str[int_file_slice.templated_slice],
+                    )
+                    templater_logger.warning(
+                        "        Python templater safety value unexpectedly triggered. "
+                        "Your query may be more conservatively sliced than expected. "
```

Comment:
I don't think the end user will know what this means. Simply a 'report this with your query' would suffice imo.